include/ccf/service/node_info_network.h

```diff
@@ -29,12 +29,14 @@ namespace ccf
   enum class ApplicationProtocol
   {
     HTTP1,
-    HTTP2
+    HTTP2,
+    CUSTOM
   };
   DECLARE_JSON_ENUM(
     ApplicationProtocol,
     {{ApplicationProtocol::HTTP1, "HTTP1"},
-     {ApplicationProtocol::HTTP2, "HTTP2"}});
+     {ApplicationProtocol::HTTP2, "HTTP2"},
+     {ApplicationProtocol::CUSTOM, "Custom"}});
```

Comment:
Custom seems more like RAW_UDP, unless I am missing something?

Reply:
I think it would be a little nicer to make application protocol a string, pass it in, and allow new protocols to register by name. We would avoid this situation where "Custom" means "a single custom thing that handles UDP", and we could keep the config & config schema unchanged.

`set_custom_protocol_subsystem(cpss)` would become `register_custom_protocol_subsystem("custom_protocol", cpss)`, what do you think @eddyashton?

Reply:
There's a different setting for TCP/UDP, but yes, "custom" here really means "application-defined". Since you can have only one per interface I figured that's enough. But you're right, it may be nicer to register them by name. Just so I understand correctly: are you thinking about allowing arbitrary strings, i.e. `ApplicationProtocol` becomes a `std::string` instead of an enum?

Reply:
I agree with @achamayou, it would be neater to put the protocol here. If I understand correctly, with the current API we'd put the protocol as the listen-interface name, and then need to match that in the app constructor? That's an awkward bit of app-code-to-config coupling, in particular making it serve a custom protocol on multiple interfaces (would need to call `cpss->install` multiple times, to match each interface name/ID). We lose type safety/early validation if we drop from enum to string lookup, but I think that's a fair tradeoff.

Reply:
All good points, I agree. If you want multiple protocols on multiple interfaces strings are definitely a better choice.

Reply:
All done, app protocol is now a string.